File: creating-non-existing-dates/ruby/invalid_times.rb
require 'date'

# Time.utc with valid input
timestamp = Time.utc(2021, 1, 15, 8, 0, 0)

puts timestamp # prints "2021-01-15 08:00:00 UTC"

# Time.utc with invalid input
timestamp = Time.utc(2021, 2, 30, 8, 0, 0)

puts timestamp # prints "2021-03-02 08:00:00 UTC"

File: creating-non-existing-dates/javascript/invalidTimes.js
// new Date with valid input
const validDate = new Date(Date.UTC(2021, 0, 15, 8, 0, 0));

console.log(validDate.toISOString()); // prints 2021-01-15T08:00:00.000Z

// new Date with invalid input
const invalidDate = new Date(Date.UTC(2021, 1, 30, 8, 0, 0));

console.log(invalidDate.toISOString()); // prints 2021-03-02T08:00:00.000Z
